app/main/posts/posts-routes.js

```diff
@@ -156,16 +163,16 @@ function (
     .state(
         {
             name: 'posts.data.detail',
-            url: '/posts/:postId',
+            url: '^/posts/:postId',
```

Comment:
Tweaked this to return to the original url of `/post/123` not `/views/data/posts/123`